packages/postcss/src/index.ts

```diff
@@ -0,0 +1,306 @@
+      }, [] as Writeable<StringifiedUtil>[])
+
+    if (!utils.length)
+      return
+
+    for (const i of utils) {
+      const [, _selector, body, parent] = i
+      const selector = _selector?.replace(regexScopePlaceholder, ' ') || _selector
+
+      if (parent || (selector && selector !== '.\\-')) {
+        const node = parse(rule.parent.toString(), {
+          context: 'rule',
+        }) as Rule
+        let newSelector = generate(node.prelude)
+        if (selector && selector !== '.\\-') {
+          const selectorAST = parse(selector, {
+            context: 'selector',
+          }) as Selector
+
+          const prelude = clone(node.prelude) as SelectorList
+
+          prelude.children.forEach((child) => {
+            const parentSelectorAst = clone(selectorAST) as Selector
+            parentSelectorAst.children.forEach((i) => {
+              if (i.type === 'ClassSelector' && i.name === '\\-')
+                Object.assign(i, clone(child))
+            })
+            Object.assign(child, parentSelectorAst)
+          })
+          newSelector = generate(prelude)
+        }
+
+        let css = `${newSelector}{${body}}`
+        if (parent)
+          css = `${parent}{${css}}`
+
+        const css_parsed = postcss.parse(css)
+        css_parsed.walkDecls((declaration) => {
+          declaration.source = source
+        })
+        rule.parent.after(css_parsed)
+      }
+      else {
+        const css = postcss.parse(body)
+        css.walkDecls((declaration) => {
+          declaration.source = source
+        })
+        rule.parent.append(css)
+      }
+    }
+
+    rule.remove()
+  })
+}
+function parseTheme(root: Root, uno: UnoGenerator, directiveName: string) {
+  const themeFnRE = new RegExp(`${directiveName}\\((.*?)\\)`, 'g')
+  root.walkDecls((decl) => {
+    const matches = Array.from(decl.value.matchAll(themeFnRE))
+
+    if (!matches.length)
+      return
+
+    for (const match of matches) {
+      const rawArg = match[1].trim()
+      if (!rawArg)
+        throw new Error(`${directiveName}() expect exact one argument, but got 0`)
+
+      let value = uno.config.theme as Record<string, any>
+      const keys = rawArg.slice(1, -1).split('.')
+
+      keys.every((key) => {
+        if (value[key] != null)
+          value = value[key]
+        else if (value[+key] != null)
+          value = value[+key]
+        else
+          return false
+        return true
+      })
+
+      if (typeof value === 'string') {
+        const code = new MagicString(decl.value)
+        code.overwrite(
+          match.index!,
+          match.index! + match[0].length,
+          value,
+        )
+        decl.value = code.toString()
+      }
+      // else if (throwOnMissing) {
+      //   throw new Error(`theme of "${rawArg.slice(1, -1)}" did not found`)
+      // }
+    }
+  })
+}
+
+interface PluginOptions {
+  content?: string[]
+  directiveMap?: {
+    apply: string
+    screen: string
+    theme: string
+  }
+}
```

Comment:
Can we have a separate type file for type only info, and re-export them. I'd also advise having more explicit name `UnoPostcssPluginOptions` or so.